src/main/java/com/google/sps/servlets/CredentialVerifierServlet.java

```diff
@@ -15,37 +17,39 @@
 
 @WebServlet("/credentials")
 public class CredentialVerifierServlet extends HttpServlet {
+  private AuthorizationCodeFlow flow;
+  private UserService userService;
+
+  @Inject
+  CredentialVerifierServlet(AuthorizationCodeFlow flow, UserService userService) {
+    this.flow = flow;
+    this.userService = userService;
+  }
+
+  public CredentialVerifierServlet() throws IOException, GeneralSecurityException {
+    this.flow = Utils.newFlow();
+    this.userService = UserServiceFactory.getUserService();
+  }
+
   @Override
   protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
-    AuthorizationCodeFlow flow;
-    try {
-      flow = Utils.newFlow();
-      response.setContentType("text/html");
-      User user = UserServiceFactory.getUserService().getCurrentUser();
-      if (user == null) {
-        response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
-        return;
-      }
-      Credential credential = flow.loadCredential(user.getUserId());
-      if (credential == null) {
-        response.getWriter().print("false");
-        return;
-      }
-      if (credential.getAccessToken() == null) {
-        if (credential.refreshToken() == false) {
-          response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-          return;
-        }
-        response.getWriter().print("true");
-        return;
-      }
-      response.getWriter().print("true");
+    response.setContentType("text/html");
+    User user = userService.getCurrentUser();
+    if (user == null) {
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+    Credential credential = flow.loadCredential(user.getUserId());
+    if (credential == null) {
+      response.getWriter().print("false");
+      return;
+    }
+    if (credential.getAccessToken() == null) {
+      response.getWriter().print("false");
```

Comment:
It could be in the same if with (credential == null || credential.getAccessToken() == null)